packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -28,6 +28,57 @@ const int _kDefaultSizeBytes = 100 << 20; // 100 MiB
 ///
 /// A shared instance of this cache is retained by [PaintingBinding] and can be
 /// obtained via the [imageCache] top-level property in the [painting] library.
+///
+/// {@tool sample}
+///
+/// This sample shows how to supply your own caching logic and replace the
+/// global [imageCache] varible.
+///
+/// ```dart
+/// import 'package:flutter/foundation.dart' as foundation;
+/// import 'package:flutter/gestures.dart' as gestures;
+/// import 'package:flutter/rendering.dart' as rendering;
+/// import 'package:flutter/scheduler.dart' as scheduler;
+/// import 'package:flutter/services.dart' as services;
+/// import 'package:flutter/widgets.dart' as widgets;
+///
+/// class MyImageCache extends widgets.ImageCache {
+///   @override
+///   void clear() {
+///     print("Clearing cache!");
+///     super.clear();
+///   }
+/// }
+///
+/// class MyWidgetsBinding extends foundation.BindingBase
```

Comment:
The example may be a little simpler by just extending WidgetsFlutterBinding?

Reply:
I think you may be right.  I was basing the sample off of what Fuchsia did.  I don't think there is any benefit to this way, maybe we should notify them this is easier.

Reply:
Switched it, done.